boards: beaglev_fire: improve docs, fix uart, expand RAM

AlexCharlton opened the pull request:
This PR makes three small changes, all aimed at improving BeagleV-Fire support.

### Update documentation
Aside from some minor clarifications, a "Flashing" section was added, since the steps to flash the Fire in a way that's compatible with the Microchip-provided [Hart Software Services](https://github.com/polarfire-soc/hart-software-services) (HSS) is not readily accomplished with `west/make flash`. An example of west scripts that are able to flash the Fire can be seen [here](https://github.com/polarfire-soc/zephyr-applications/), but I doubt this ought to live in Zephyr proper.

### Change selected UART
In https://github.com/zephyrproject-rtos/zephyr/commit/166b9bf35acc3a6c8a3665c11fa6e9b7054506c0#diff-01f5d360c1043dce50bf74fbb1cc056d2049160741e483f843c19d081c065252R18 the selected UART was changed from `uart0` to `uart1`, presumably to mirror the changes made to the other PolarFire boards. While this makes sense to do for the Icicle kit, from looking at the [Fire's schematics](https://git.beagleboard.org/beaglev-fire/beaglev-fire/-/blob/main/BeagleV-Fire_sch.pdf), I don't believe any of the UARTs other than `uart0` is hooked up to anything. 

### Expand available RAM
`sram0` is a relatively small region of RAM, meant to be used by the HSS (see [Linux device tree](https://github.com/linux4sam/linux-at91/blob/bf96df830936986fdb3c7789749ff599300dae01/arch/riscv/boot/dts/microchip/mpfs-beaglev-fire.dts#L73)). I see no reason why `sram1` shouldn't occupy the remaining ~2GB present on the Fire. I'm a Zephyr newbie, however, so I'm not sure if there are any reasons why you would want to limit the RAM defined.

AlexCharlton committed:
boards: beaglev_fire: document flashing board

In addition to some minor corrections, document the process of flashing
the board.

Signed-off-by: Alex Charlton <alex.n.charlton@gmail.com>

AlexCharlton committed:
boards: beaglev_fire: fix uart

uart1 is not connected to anything as far as I can tell.
uart0 was the previous correctly selected uart, so this changes
back to that.

Signed-off-by: Alex Charlton <alex.n.charlton@gmail.com>

AlexCharlton committed:
boards: beaglev_fire: expand RAM

The Fire has 2GB of RAM, so expand sram1 to use it

Signed-off-by: Alex Charlton <alex.n.charlton@gmail.com>

AlexCharlton commented:
Noting that these doc fixes overlap with https://github.com/zephyrproject-rtos/zephyr/pull/79511. My doc additions do cover the flashing instructions which were noted as missing in that PR.

con-pax commented:
Hey @AlexCharlton, thanks for the PR! It all looks good to me. I will test it on the Beagle today.
Cheers

AlexCharlton commented:
Thanks @con-pax! I should have noted that I tested this out on my Fire, and it Works For Me 🙂

con-pax reviewed (dismissed):
LGTM
```bash
6.406446] >> *** Booting Zephyr OS build v4.0.0-rc1-79-g464e68645d7f ***
Hello World! beaglev_fire/polarfire/e51
```

```bash
464e68645d7 (HEAD -> main, origin/main, origin/HEAD) boards: beaglev_fire: expand RAM
```

```bash
pc             0x80000290	0x80000290 <main+24>
```

Ayush1325 commented on boards/beagle/beaglev_fire/doc/index.rst:
```diff
@@ -15,11 +15,17 @@ hobbyists, and researchers to explore and experiment with RISC-V technology.
 .. zephyr-app-commands::
    :zephyr-app: samples/hello_world
-   :board: beaglev_fire
+   :board: beaglev_fire_polarfire_u54
```
The board name should be `beaglev_fire/polarfire/u54`

AlexCharlton replied:
Yes! Good catch 🙏

AlexCharlton replied:
Fixed

AlexCharlton committed:
boards: beaglev_fire: fix board identifiers in docs

Per PR review.

Signed-off-by: Alex Charlton <alex.n.charlton@gmail.com>

AlexCharlton committed:
boards: beaglev_fire: fix memory address

Per the docs, the memory at address 0x80000000 ends at 0xC0000000.
In other words, the address space is 0x40000000, which is only half
of the size we want to map. This means that the upper address space
previously mapped was overlapping with the space reserved for non-cached
memory.

Instead, we map the entire 2GB at 0x1000000000, which is the correct
address for cached DDR that occupies more than 1 GB.

We defined a new node in the device tree for this memory region,
`beaglev.ddr_cached_high`. We did not reuse the `soc` node because
we needed to redefine the `#address-cells` to be 2, and doing so
would have affected other nodes under `soc`.

Signed-off-by: Alex Charlton <alex.n.charlton@gmail.com>

kartben commented:
@jadonk we still don't have you properly set up as a maintainer, apparently...

kartben commented:
rebased to re-trigger incomplete CI
